onnx_caffe2/backend.py

```diff
@@ -197,6 +200,33 @@ def _create_constant(cls, n):
         assert len(n.outputs) == 1
         return cls._create_tensor_filling_op(n.attrs["value"], n.outputs[0])
 
+    @classmethod
+    def _create_gemm(cls, n):
+        (A, B, C) = n.inputs
+        (Y,) = n.outputs
+        alpha = n.attrs.get('alpha', 1)
+        beta = n.attrs.get('beta', 1)
```

Comment:
It may be worth adding a special case translation to FC, but this will do for now...

Reply:
Tried that, the broadcst thingy really makes it complicated, caffe2 FC only accepts one dimensional C.

Reply:
So, I was under the impression that `alpha == 1 and beta == 1 and broadcast == True` was the necessary and sufficient condition to use FC. Not the case?

Reply:
alpha and beta needs to be handled separated anyway (no matter we use "FC" or "MatMul" + "Add"). For `broadcast=True`, FC only accepts `C` being 1d. I thought in onnx the constraint is more relaxed (e.g. A is 4x4 and C can be 3x2)?

Reply:
OK, hang on...

So, looking at this code again, we don't have handling for the transpositions which are supported in the spec, which we should add.

I'm not sure exactly what rules should result in choosing FC versus MatMul, but it would be good if we could avoid redundant transposes (so we'll probably need some sort of "transpose" analysis pass), and use the fused broadcasted addition when it is available. It's probably worth doing this properly.

Reply:
Transpose is handled in MatMul

Reply:
Oh right. So I guess, the only case for when you'd use FC is when you have a nonzero bias and the transposes line up exactly. (Dunno which is more expensive: transpose before, or unfused bias addition after.)

Reply:
my guess is transpose is more expensive because of the memory access pattern.

Reply:
I'm not sure how things work under the hood in Caffe2, but standard BLAS allows you to specify if the matrices are row or column major (transposed or not), so if Caffe2 follows the same pattern, multiplying by the transpose of a matrix should be fairly cost-free.